### PR TITLE
Added support for select2 tags

### PR DESCRIPTION
**Pull request by micbis**
## Subject

Added support for select2 tags (to not only allow choosing existing values, but also specifying a new one)

I am targeting this branch, because is a feature addition to the current released branch (4.x)

## Changelog

```markdown
### Added

- Support for select2 tags option
```

### CODE REVIEW

File: assets/js/admin.js [hunk 1]
```diff
@@ -101,6 +102,12 @@ const Admin = {
           allowClearEnabled = false;
         }
 
+        if (select.attr('data-sonata-select2-allow-tags') === 'false') {
+          allowTags = false;
+        } else {
+          allowTags = true;
+        }
```
Comment (VincentLanglet) [inline assets/js/admin.js]: By default `data-sonata-select2-allow-tags` won't be set so it won't be false, so you set allowTags as true.

It should stay false by defaut no ?

**Review by VincentLanglet — CHANGES_REQUESTED** [review]

File: assets/js/admin.js (hunk 1 above)
Comment (micbis) [reply assets/js/admin.js]: Good catch! We updated the pr to keep the default behaviour.

File: docs/cookbook/recipe_select2.rst [hunk 2]
```diff
@@ -71,6 +71,29 @@ to enable ``allowClear`` or ``data-sonata-select2-allow-clear = "false"`` to dis
...
+                'attr' => [
+                    'data-sonata-select2-allow-tags' => 'true'
+                ]
+            ])
+        ;
+    }
+
+.. note::
+
+    You have to use false as string! ``"false"``!
```
Comment (VincentLanglet) [inline docs/cookbook/recipe_select2.rst]:
```suggestion
    You have to use true as string! ``"true"``!
```
then

**Comment by VincentLanglet** [comment]
Hi @micbis You need to compile the assets and commit it too. (cf failing build)

Something like
```
npm install
npm run encore production
```

**Review by VincentLanglet — APPROVED** [review]

**Comment by VincentLanglet** [comment]
Thanks

**Pull request merged**